x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/time_shift.tsx

```diff
@@ -0,0 +1,271 @@
+      defaultMessage: '12 hours (12h)',
+    }),
+    value: '12h',
+  },
+  {
+    label: i18n.translate('xpack.lens.indexPattern.timeShift.day', {
+      defaultMessage: '1 day (1d)',
+    }),
+    value: '1d',
+  },
+  {
+    label: i18n.translate('xpack.lens.indexPattern.timeShift.week', {
+      defaultMessage: '1 week (1w)',
+    }),
+    value: '1w',
+  },
+  {
+    label: i18n.translate('xpack.lens.indexPattern.timeShift.month', {
+      defaultMessage: '1 month (1M)',
+    }),
+    value: '1M',
+  },
+  {
+    label: i18n.translate('xpack.lens.indexPattern.timeShift.3months', {
+      defaultMessage: '3 months (3M)',
+    }),
+    value: '3M',
+  },
+  {
+    label: i18n.translate('xpack.lens.indexPattern.timeShift.6months', {
+      defaultMessage: '6 months (6M)',
+    }),
+    value: '6M',
+  },
+  {
+    label: i18n.translate('xpack.lens.indexPattern.timeShift.year', {
+      defaultMessage: '1 year (1y)',
+    }),
+    value: '1y',
+  },
+  {
+    label: i18n.translate('xpack.lens.indexPattern.timeShift.previous', {
+      defaultMessage: 'Previous',
+    }),
+    value: 'previous',
+  },
+];
+
+export function TimeShift({
+  selectedColumn,
+  columnId,
+  layer,
+  updateLayer,
+  indexPattern,
+  isFocused,
+  activeData,
+  layerId,
+}: {
+  selectedColumn: IndexPatternColumn;
+  indexPattern: IndexPattern;
+  columnId: string;
+  layer: IndexPatternLayer;
+  updateLayer: (newLayer: IndexPatternLayer) => void;
+  isFocused: boolean;
+  activeData: IndexPatternDimensionEditorProps['activeData'];
+  layerId: string;
+}) {
+  const [localValue, setLocalValue] = useState(selectedColumn.timeShift);
+  useEffect(() => {
+    setLocalValue(selectedColumn.timeShift);
+  }, [selectedColumn.timeShift]);
+  const selectedOperation = operationDefinitionMap[selectedColumn.operationType];
+  if (!selectedOperation.shiftable || selectedColumn.timeShift === undefined) {
+    return null;
+  }
+
+  let dateHistogramInterval: null | moment.Duration = null;
+  const dateHistogramColumn = layer.columnOrder.find(
+    (colId) => layer.columns[colId].operationType === 'date_histogram'
+  );
+  if (!dateHistogramColumn && !indexPattern.timeFieldName) {
+    return null;
+  }
+  if (dateHistogramColumn && activeData && activeData[layerId] && activeData[layerId]) {
+    const column = activeData[layerId].columns.find((col) => col.id === dateHistogramColumn);
+    if (column) {
+      dateHistogramInterval = search.aggs.parseInterval(
+        search.aggs.getDateHistogramMetaDataByDatatableColumn(column)?.interval || ''
+      );
+    }
+  }
+
+  function isValueTooSmall(parsedValue: ReturnType<typeof parseTimeShift>) {
+    return (
+      dateHistogramInterval &&
+      parsedValue &&
+      typeof parsedValue === 'object' &&
+      parsedValue.asMilliseconds() < dateHistogramInterval.asMilliseconds()
+    );
+  }
+
+  function isValueNotMultiple(parsedValue: ReturnType<typeof parseTimeShift>) {
```

Comment:
This check isn't applied to the custom values, so I can set my date histogram to 17 minutes and attempt to shift by 30 minutes- this should not work.